deploy/src/main/resources/README.md

```diff
@@ -10,106 +10,168 @@ The Job Service is a RESTful Web Service and provides a simple API.  It tracks t
+        <td>CAF_WORKER_GLOBFILTER_INPUT_QUEUE</td>
+        <td>The RabbitMQ queue on which the Glob Filter worker listens.</td>
+      </tr>
+      <tr>
+        <td>CAF_BATCH_WORKER_ERROR_QUEUE</td>
+        <td>The RabbitMQ queue where failed Glob Filter worker messages are sent.</td>
+      </tr>
+      <tr>
+        <td>CAF_GLOB_WORKER_BINARY_DATA_INPUT_FOLDER</td>
+        <td>The location of the mounted directory inside the container where the test files are located.</td>
+      </tr>
+      <tr>
+        <td>CAF_WORKER_LANGDETECT_INPUT_QUEUE</td>
+        <td>The RabbitMQ queue on which the Language Detection worker listens.</td>
+      </tr>
+      <tr>
+        <td>CAF_WORKER_LANGDETECT_OUTPUT_QUEUE</td>
+        <td>The RabbitMQ queue on which the Language Detection worker outputs messages.</td>
+      </tr>
+      <tr>
+        <td>CAF_LANG_DETECT_WORKER_OUTPUT_FOLDER</td>
+        <td>The folder in which the Language Detection worker places result files.</td>
       </tr>
       <tr>
         <td>JOB_SERVICE_DEMO_INPUT_DIR</td>
-        <td>./input&#8209;files</td>
-        <td>This directory is made available as a source for input files which may be read. The glob-pattern is passed in as a parameter but this is the base directory that it starts from; it cannot read any files which are outside this directory.</td>
+        <td>The directory where the test files are located on the host.</td>
       </tr>
       <tr>
         <td>JOB_SERVICE_DEMO_OUTPUT_DIR</td>
-        <td>./output&#8209;files</td>
-        <td>This directory is used for storing the output from the Language Detection operation.</td>
+        <td>The output directory for test results on the host.</td>
       </tr>
     </table>
 
-    In order to run multiple instances of the demonstration stack simultaneously it would be necessary to set the `JOB_SERVICE_PORT` parameter to different values for each instance.
+4. Deploy the services  
+	Deploy the persistent volume by issuing the following command from the directory where you have downloaded the files to:
 
-3. Deploy the services  
-    First navigate to the folder where you have downloaded the files to and then run one of the following commands, depending on whether you are using Docker Compose or Docker Stack:
+    `kubectl create -f jobservice-persistentvolumeclaim.yaml`
 
-    <table>
-      <tr>
-        <td><b>Docker Compose</b></td>
-        <td>docker-compose up</td>
-      </tr>
-      <tr>
-        <td><b>Docker Stack</b></td>
-        <td>docker stack deploy --compose-file=docker-compose.yml jobservicedemo</td>
-      </tr>
-    </table>
+	Deploy the database and RabbitMQ by issuing the following command from the directory where you have downloaded the files to:
+
+    ```
+	source ./kubernetes.env \
+            ; cat jobservice-deployment.yaml \
+            | perl -pe 's/\$\{(\w+)\}/(exists $ENV{$1} && length $ENV{$1} > 0 ? $ENV{$1} : "NOT_SET_$1")/eg' \
+            | kubectl create -f -
+	```
```

Comment:
We had to do something like this for Marathon, but doesn't the `kubectl` command have in-built facilities for this?  What does `kubectl apply` do for example?

Reply:
The help states "Apply a configuration to a resource by filename or stdin."

At this stage I do not know enough about the kubectl to implement this. I just copied the same method as marathon as it works. I could possibly look into other nicer methods but I do not have time to do this before the end of sprint.